### ISSUE TITLE
INSTALL_PREFIX vs INSTALLED_PREFIX in configuration.c

### ISSUE THREAD

**Issue by nkurz** [RRZE-HPC/likwid#33]
I don't know if this is really an issue, and I haven't figured out exactly what it is doing, but it looks suspicious.     In configuration.c, it checks for likwid.cfg and likwid_topo.cfg in INSTALL_PREFIX rather than in INSTALLED_PREFIX.  Is this correct?  I'm compiling for MIC and have these set differently.

**Comment by nkurz** [comment]
OK, I've dug in deeper and solved the issue I was aiming for.    I was having problems running the binaries on Phi because only some of the prefixes were being changed.   I don't think that changing changing configuration.c is necessary, rather the changes needed to be made to the Makefiles and includes.   I've attached a patch that seems to work for me, although I've only tested it on a single system so far.  

[likwid_prefix.txt](https://github.com/RRZE-HPC/likwid/files/252982/likwid_prefix.txt)

**Comment by TomTheBear** [comment]
We had never problems with the current prefixes. The installed_prefix stuff was integrated through a pull request to support packaging of LIKWID where the build root and the final position is different. I check your changes on some systems including Xeon Phi coprocessors to verify your changes and the old behavior.

**Comment by TomTheBear** [comment]
I tested the paths thing and did some refinements when PREFIX is not equal to INSTALLED_PREFIX. I tested it on a Xeon Phi and had no problems with it.
I added a section on the [Build instructions](https://github.com/RRZE-HPC/likwid/wiki/Build#installation-on-intel-xeon-phi) wiki page for Xeon Phi, maybe it helps you.

Commonly, you only have to set `PREFIX` to some location and use `make && (sudo) make install`. The `INSTALLED_PREFIX` is only interesting if you have something like a temporary install location (e.g. the build root of packaging solutions) which is not the final location. I checked that anything internal depends on `INSTALLED_PREFIX` (internally named `INSTALL_PREFIX`).

If you have time, please try the current LIKWID master branch and give me some response how it worked out for you.

**Comment by nkurz** [comment]
Nice work!  Yes, I just tested, and it works for me.   I've attached a patch for a few other changes to make it smoother.   Some of them are specific to my system, but seemed worth providing a full diff for completeness. 

1) I removed "-Wno-unused-result" from LFLAGS in hwloc/Makefile because ICC doesn't recognize it.  If it's useful, it could be put back with a conditional, although LFLAGS seems like an unlikely place for it. 

2) I downgraded your warning in config_checks.mk to ignore MIC.   I think it's more confusing then helpful in cases where one is cross-compiling.  Or maybe just change to "Info" rather than "Warning"? 

3) In config_defines.mk I changed "Set accessmode" to "Changing accessmode".  "Set" is both the past tense, and the imperative.  When I first read the message, I thought it was telling me to change the accessmode, whereas it had already done it for me.  I think this is clearer. 

4) I think you were doubling up the RPATHS in the version you had.  I think this is better split up this way.  

I should also explain the use case.  I'm installing compiling for Phi on a system that does not have a shared NFS system.    I set prefix to install into "/var/mpss/common/usr", which is an overlay directory used to create the file system on the Phi when it is initialized.    On the Phi, this directory automatically becomes "/usr", hence the need for a distinction between PREFIX and INSTALLED_PREFIX.

Thanks for making this work!  

[likwid_phi.txt](https://github.com/RRZE-HPC/likwid/files/258290/likwid_phi.txt)
